packages/@uppy/core/src/Restricter.js

```diff
@@ -30,16 +30,37 @@ class Restricter {
     }
   }
 
-  validate (file, files) {
-    const { maxFileSize, minFileSize, maxTotalFileSize, maxNumberOfFiles, allowedFileTypes } = this.getOpts().restrictions
+  validateTotals (existingFiles, newFiles) {
```

Comment:
can we rename this to something more intention revealing? `validateSizeAndCount`?

Reply:
But then if we add some more aggregate validation to it, we have to rename it again

Reply:
Sounds like a premature optimisation? We can't write code for a future we can't predict.

Reply:
I'm pretty sure we can predict that if we ever implement another aggregate limitation (limit on sum of files), it will be put inside this function, because this function operates on all files, as opposed to the other one

Reply:
For me there are no exceptions to premature optimisations if there are not already defined and planned. "If we ever" is a textbook example of this problem. It doesn't have to be my suggestion for the name, but `Totals` gives no information in the present.

Reply:
it's not so much an optimization of the code, it's just a naming. I feel that we should give the function a name that describes in a generic way what it does. maybe totals is not the best name. What about validateAggregateRestrictions?

Reply:
Sounds good!